backend/frontend/tests/test_card_done_button.py

```diff
@@ -0,0 +1,101 @@
+from activity_log.models import LogEntry
+from core.tests.factories import UserFactory
+from curriculum_tracking.tests.factories import AgileCardFactory, ContentItemFactory
+from curriculum_tracking.models import ContentItem, AgileCard
+import curriculum_tracking.activity_log_entry_creators as creators
+from .frontend_test_mixin import FrontendTestMixin
+
+
+class TestCardDoneButton(FrontendTestMixin):
+    def setUp(self):
+        super().setUp()
+        self.user = UserFactory()
+        self.user.set_password(self.user.email)
+        self.user.save()
+
+        self.do_login(self.user)
+
+    
+    def make_topic_card(self):
+        self.card: AgileCard = AgileCardFactory(
+            content_item=ContentItemFactory(
+                content_type=ContentItem.TOPIC,
+            ),
+            status=AgileCard.READY,
+        )
+        self.card.assignees.set([self.user])
+        self.card.start_topic()
+    
+    def make_project_card(self):
+        self.card: AgileCard = AgileCardFactory(
+            content_item=ContentItemFactory(
+                content_type=ContentItem.PROJECT,
+                project_submission_type=ContentItem.LINK,
+            ),
+            status=AgileCard.IN_PROGRESS,
+        )
+        self.card.assignees.set([self.user])
+
+    def make_outstanding_ir_project_card(self):
+        self.card: AgileCard = AgileCardFactory(
+            content_item=ContentItemFactory(
+                content_type=ContentItem.PROJECT,
+                project_submission_type=ContentItem.LINK,
+            ),
+            status=AgileCard.IN_REVIEW,
+        )
+        self.card.reviewers.set([self.user])
+
+    def test_done_button_does_not_show_for_project_cards(self):
+        self.make_project_card()
+        ip_column = self.page.text_content("div#column_IP")
+        
+        self.assertNotIn("Done", ip_column)
+
+    def test_done_button_shows_for_topic_cards(self):
+        self.make_topic_card()
+        ip_topic_card = self.page.text_content("div#column_IP")
```

Comment:
This tests passes and fails sometimes:
```
FAIL: test_done_button_shows_for_topic_cards (frontend.tests.test_card_done_button.TestCardDoneButton)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ngoako/Desktop/code/umuzi/Tilde/backend/frontend/tests/test_card_done_button.py", line 59, in test_done_button_shows_for_topic_cards
    self.assertIn("Done", ip_topic_card)
AssertionError: 'Done' not found in ''

----------------------------------------------------------------------
Ran 5 tests in 16.595s

FAILED (failures=1)
Destroying test database for alias 'default'...
```

Waiting before selecting seems to resolve it:
```
self.page.wait_for_load_state("networkidle")
```

Reply:
Still failing occasionally, You have to wait for everything to load before selecting the column, not after the selection of ip_column.